// Copyright 2010, Tencent Inc.
// Author: fatliu (fatliu@tencent.com)
//
// Define protocol buffers for CA identity, role, relation and quota.
package ca;

message IdNameRecord {
	required uint32 id = 1;
	required string name = 2;
	required bool valid = 3;
};

message RelationRecord {
	required uint32 iid = 1;
	required uint32 rid = 2;
	required bool valid = 3;
};

message QuotaStorageRecord {
	required uint32 rid = 1;
	required string cluster = 2;
	required uint32 num_chunks = 3;
	required uint32 num_files = 4;
	required uint32 num_dirs = 5;
};

message IdentityList {
	repeated IdNameRecord identity = 1;
}

message RoleList {
	repeated IdNameRecord role = 1;
}

message RelationList {
	repeated RelationRecord relation = 1;
}

message QuotaList {
	repeated QuotaStorageRecord quota = 1;
}
